### PR TITLE
Fixes to `evaluate` standard error calculation to fix display issue

### LINKED ISSUE #871: `evaluate` errors displaying when using `confusion_matrix` as measure
When using confusion matrix as a measure in `evaluate` or in `evaluate!` I get this error with displaying the aggregated confusion matrix. After checking, I found the result does have the correct confusion matrix. But the full results cannot be displayed, only each individual part separately. 

Minimal example

```
using MLJ
import DataFrames

X = DataFrames.DataFrame(
    :x1 => rand(100),
    :x2 => rand(100),
);

y = randn(100) .> 0;
y = coerce(y, OrderedFactor);

@load LogisticClassifier pkg=MLJLinearModels

mdl = MLJLinearModels.LogisticClassifier()
cv = CV(nfolds=5)

@show scitype(X) # Table{AbstractVector{Continuous}}

@show aggregation(confusion_matrix) # StatisticalTraits.Sum()

result = evaluate(mdl, X, y, resampling=cv, measures=confusion_matrix)

result.measurement[1] # does hold the confusion matrix
```

the last line gives this error:
```
ERROR: MethodError: no method matching /(::MLJBase.ConfusionMatrixObject{2}, ::Int64)
Closest candidates are:
  /(::Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}, ::Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}) at int.jl:97
  /(::Union{SparseArrays.SparseVector{Tv, Ti}, SubArray{Tv, 1, <:SparseArrays.AbstractSparseMatrixCSC{Tv, Ti}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, false}, SubArray{Tv, 1, <:SparseArrays.AbstractSparseVector{Tv, Ti}, Tuple{Base.Slice{Base.OneTo{Int64}}}, false}} where {Tv, Ti}, ::Number) at ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/SparseArrays/src/sparsevector.jl:1519
  /(::Union{ColorTypes.LMS, ColorTypes.XYZ}, ::Number) at ~/.julia/packages/Colors/yDxFN/src/algorithms.jl:9
  ...
Stacktrace:
  [1] _mean(f::typeof(identity), A::Vector{MLJBase.ConfusionMatrixObject{2}}, dims::Colon)
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:176
  [2] mean(A::Vector{MLJBase.ConfusionMatrixObject{2}}; dims::Function)
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:164
  [3] mean(A::Vector{MLJBase.ConfusionMatrixObject{2}})
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:164
  [4] _var(A::Vector{MLJBase.ConfusionMatrixObject{2}}, corrected::Bool, mean::Nothing, #unused#::Colon)
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:379
  [5] var(A::Vector{MLJBase.ConfusionMatrixObject{2}}; corrected::Bool, mean::Nothing, dims::Function)
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:368
  [6] _std(A::Vector{MLJBase.ConfusionMatrixObject{2}}, corrected::Bool, mean::Nothing, #unused#::Colon)
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:454
  [7] std(A::Vector{MLJBase.ConfusionMatrixObject{2}}; corrected::Bool, mean::Nothing, dims::Function)
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:449
  [8] std(A::Vector{MLJBase.ConfusionMatrixObject{2}})
    @ Statistics ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/Statistics/src/Statistics.jl:449
  [9] (::MLJBase.var"#291#292"{Int64, Float64})(per_fold::Vector{MLJBase.ConfusionMatrixObject{2}})
    @ MLJBase ~/.julia/packages/MLJBase/LxNAs/src/resampling.jl:543
 [10] iterate
    @ ./generator.jl:47 [inlined]
 [11] _collect
    @ ./array.jl:807 [inlined]
 [12] collect_similar
    @ ./array.jl:716 [inlined]
 [13] map
    @ ./abstractarray.jl:2933 [inlined]
 [14] _standard_errors
    @ ~/.julia/packages/MLJBase/LxNAs/src/resampling.jl:542 [inlined]
 [15] show(io::IOContext{Base.TTY}, #unused#::MIME{Symbol("text/plain")}, e::PerformanceEvaluation{Vector{ConfusionMatrix}, Vector{MLJBase.ConfusionMatrixObject{2}}, Vector{typeof(predict_mode)}, Vector{Vector{MLJBase.ConfusionMatrixObject{2}}}, Vector{Missing}, Vector{NamedTuple{(:classes, :coefs, :intercept), Tuple{CategoricalArrays.CategoricalVector{Bool, UInt32, Bool, CategoricalArrays.CategoricalValue{Bool, UInt32}, Union{}}, Vector{Pair{Symbol, Float64}}, Float64}}}, Vector{Nothing}})
    @ MLJBase ~/.julia/packages/MLJBase/LxNAs/src/resampling.jl:552
 [16] (::REPL.var"#43#44"{REPL.REPLDisplay{REPL.LineEditREPL}, MIME{Symbol("text/plain")}, Base.RefValue{Any}})(io::Any)
    @ REPL ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/REPL/src/REPL.jl:267
 [17] with_repl_linfo(f::Any, repl::REPL.LineEditREPL)
    @ REPL ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/REPL/src/REPL.jl:521
 [18] display(d::REPL.REPLDisplay, mime::MIME{Symbol("text/plain")}, x::Any)
    @ REPL ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/REPL/src/REPL.jl:260
 [19] display(d::REPL.REPLDisplay, x::Any)
    @ REPL ~/.julia/juliaup/julia-1.8.3+0.x64/share/julia/stdlib/v1.8/REPL/src/REPL.jl:272
 [20] display(x::Any)
    @ Base.Multimedia ./multimedia.jl:328
 [21] #invokelatest#2
    @ ./essentials.jl:729 [inlined]
 [22] invokelatest
    @ ./essentials.jl:726 [inlined]
 [23] (::VSCodeServer.var"#66#70"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/eval.jl:199
 [24] withpath(f::VSCodeServer.var"#66#70"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams}, path::String)
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/repl.jl:249
 [25] (::VSCodeServer.var"#65#69"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/eval.jl:155
 [26] hideprompt(f::VSCodeServer.var"#65#69"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/repl.jl:38
 [27] (::VSCodeServer.var"#64#68"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/eval.jl:126
 [28] with_logstate(f::Function, logstate::Any)
    @ Base.CoreLogging ./logging.jl:511
 [29] with_logger
    @ ./logging.jl:623 [inlined]
 [30] (::VSCodeServer.var"#63#67"{VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/eval.jl:225
 [31] #invokelatest#2
    @ ./essentials.jl:729 [inlined]
 [32] invokelatest(::Any)
    @ Base ./essentials.jl:726
 [33] macro expansion
    @ ~/.vscode/extensions/julialang.language-julia-1.38.2/scripts/packages/VSCodeServer/src/eval.jl:34 [inlined]
 [34] (::VSCodeServer.var"#61#62")()
    @ VSCodeServer ./task.jl:484
```

### PR DESCRIPTION
**Pull request by ablaom**
Closes #871.

### CODE REVIEW

File: test/resampling.jl [hunk 1]
```diff
@@ -801,8 +802,22 @@ end
...
+        clf,
+        X,
+        y,
+        operation=predict_mode,
+        measure = ConfusionMatrix(),
+        resampling=CV(),
+    )
+    io = IOBuffer()
+    show(io, evaluations)
+    show(io, MIME("text/plain"))
```
Comment (rikhuijzer) [inline test/resampling.jl:819]:
```suggestion
    confusion_table = sprint(show, "text/plain", evaluations.measurement[1])
    @test contains(confusion_table, "Ground Truth")
```
I've also tried to test for "N/A" being contained in `display(evaluations)`, but was unable for some reason:
```julia
julia> display(evaluations)
PerformanceEvaluation object with these fields:
  measure, operation, measurement, per_fold,
  per_observation, fitted_params_per_fold,
  report_per_fold, train_test_rows
Extract:
┌───────────────────┬──────────────┬───────────────────────────────────────────
│ measure           │ operation    │ measurement                              ⋯
├───────────────────┼──────────────┼───────────────────────────────────────────
│ ConfusionMatrix(  │ predict_mode │ ConfusionMatrixObject{2}([20 30; 30 20], ⋯
│   perm = nothing) │              │                                          ⋯
└───────────────────┴──────────────┴───────────────────────────────────────────
                                                              3 columns omitted

julia> sprint(show, "text/plain", evaluations)
"PerformanceEvaluation object with these fields:\n  measure, operation, measurement, per_fold,\n  per_observation, fitted_params_per_fold,\n  report_per_fold, train_test_rows\nExtract:\n┌───────────────────┬──────────────┬───────────────────────────────────────────────────" ⋯ 4277 bytes ⋯ "────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘\n"
```
For some reason, the `PrettyTables.jl` output appears to be truncated. Do you maybe know how to circumvent this? This is optional of course. The PR looks good to me
```

Comment (ablaom) [reply test/resampling.jl:819]: Your suggestion is really an *additional* test. It won't catch the original fail, which only happens if you try to display the full PerformanceEvaluation object, right? 

I don't know how to handle the truncation. Was hoping https://github.com/ronisbr/PrettyTables.jl/issues/87 would happen, but that's been a while. Happy for suggestions, but we should merge this to address the bug.

**Review by rikhuijzer — APPROVED** [review]

File: test/resampling.jl [hunk 2]
```diff
@@ -801,8 +802,23 @@ end
...
+        y,
+        operation=predict_mode,
+        measure=ConfusionMatrix(),
+        resampling=CV(),
+    )
+    io = IOBuffer()
+    show(io, evaluations)
+    show(io, MIME("text/plain"))
+    confusion_table = sprint(show, "text/plain", evaluations.measurement[1])
+    @test contains(confusion_table, "Ground Truth")
```
Comment (rikhuijzer) [inline test/resampling.jl:821]:
```suggestion
    printed_evaluations = sprint(show, "text/plain", evaluations)
    @test contains(printed_evaluations, "N/A")
```

This catches the failure reported in #871. My `sprint(show, "text/plain", evaluations.measurement[1])` test didn't so can be removed.

**Review by rikhuijzer — CHANGES_REQUESTED** [review]

File: test/resampling.jl (hunk 2 above)
Comment (rikhuijzer) [reply test/resampling.jl:821]: With old code in `src/resampling` (without this PR):
```julia
julia> sprint(show, "text/plain", evaluations)
ERROR: MethodError: no method matching /(::MLJBase.ConfusionMatrixObject{2}, ::Int64)
[...]
```
With new code in `src/resampling` (with this PR):
```julia
julia> sprint(show, "text/plain", evaluations) |> print
PerformanceEvaluation object with these fields:
  measure, operation, measurement, per_fold,
  per_observation, fitted_params_per_fold,
  report_per_fold, train_test_rows
Extract:
┌───────────────────┬──────────────┬──────────────────────────────────────────────────────┬─────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ measure           │ operation    │ measurement                                          │ 1.96*SE │ per_fold                                                                                                                                                                                                                                                                                                                                           │
├───────────────────┼──────────────┼──────────────────────────────────────────────────────┼─────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ ConfusionMatrix(  │ predict_mode │ ConfusionMatrixObject{2}([18 32; 32 18], ["0", "1"]) │ N/A     │ MLJBase.ConfusionMatrixObject{2}[ConfusionMatrixObject{2}([6 11; 0 0], ["0", "1"]), ConfusionMatrixObject{2}([0 0; 10 7], ["0", "1"]), ConfusionMatrixObject{2}([7 10; 0 0], ["0", "1"]), ConfusionMatrixObject{2}([0 0; 10 7], ["0", "1"]), ConfusionMatrixObject{2}([5 11; 0 0], ["0", "1"]), ConfusionMatrixObject{2}([0 0; 12 4], ["0", "1"])] │
│   perm = nothing) │              │                                                      │         │                                                                                                                                                                                                                                                                                                                                                    │
└───────────────────┴──────────────┴──────────────────────────────────────────────────────┴─────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Comment (ablaom) [reply test/resampling.jl:821]: I stand corrected. Thanks for the review and taking the time to sort out my misunderstanding.

**Comment by ablaom** [comment]
@rikhuijzer Thank you for reviewing and improving this PR.

**Comment by rikhuijzer** [comment]
> @rikhuijzer Thank you for reviewing and improving this PR.

Thank you for your work on MLJ :heart:

**Pull request merged**